### PR TITLE
bus_rebuild: rebuild internal nets, not just ports

### PR DESCRIPTION
**Pull request by akashlevy**
## Summary

`bus_rebuild` only ever scanned wires with `port_input`/`port_output` set, so bit-blasted internal nets stayed blasted. Those are the nets that matter for characterization: since the surround-with-flops methodology, the cell-under-test hangs off the flop boundary nets (`<cell>_<pin>__pqi` / `__pqo`) rather than off the enclosing module's ports, and those nets are what `carvenetlist` turns into the carved cell's pins.

Reproduced on the Preqorsor side with a two-bit registered cell input: the carved cell came out with a single 1-bit port `fast_<enc>_B_0_` — bit 1 dropped outright — instead of `fast_<enc>_B[1:0]`. A pin named `B_0_` is then mangled to `B__` when the characterized Liberty is read back, so the arcs and input caps go missing and `chklib` reports the reference cell as uncharacterized.

The rewrite machinery needed no change: `rewrite_sigspecs` plus `remove` already repoint cells and connections alike, so only the wire scan had to widen. Four guards on top:

- `$`-prefixed autogenerated names are never grouped, since no writer bit-blasted those.
- A group mixing ports with internal nets is dropped whole, so a net cannot be absorbed into a port's bus and silently widen the module's interface.
- Only port groups reach `reconnect_instances`; an internal net has no formal on an instance to merge.
- `\keep` is carried onto the rebuilt bus. A port is kept by virtue of being a port, but a net that loses `keep` becomes removable, so merging would otherwise let `opt_clean` sweep it away.

This is what the pass's own help text already claimed it did; the help and the dev docs are updated to describe the guards accurately.

## Test plan

Three cases added to `tests/silimate/bus_rebuild.ys`:

- [x] A plain bit-blasted internal net groups into a bus, with `\keep` surviving `opt -full`
- [x] The flop boundary case (`__pqi` / `__pqo` across `$dff`), proved with `equiv_opt -assert bus_rebuild` so equivalence holds through the flops
- [x] A group mixing a port with a net is rejected, checked via `logger -expect warning`
- [x] The nine pre-existing port cases still pass unchanged
- [x] `tests/silimate/carvenetlist.ys` passes

Preqorsor side (companion PR): the golden `ref_reconstruct.json` for `splitnetlist` is byte-identical, so port behavior is untouched, and a new end-to-end test runs the real flow order (`flatten` -> `bus_rebuild` -> `carvenetlist`) on a fully bit-blasted train netlist. That test fails against the current wheel and passes against this branch.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: passes/silimate/bus_rebuild.cc [hunk 1]
```diff
@@ -188,6 +208,12 @@ struct BusRebuildPass : public Pass {
 			bus->port_id = port_id;
 			bus->set_src_attribute(members.begin()->second->get_src_attribute());
 
+			// A net carrying \keep is one opt_clean was told to leave alone. A port is kept by
+			// virtue of being a port, but a net would quietly become removable once merged.
+			for (auto &member : members)
+				if (member.second->get_bool_attribute(ID::keep))
+					bus->set_bool_attribute(ID::keep);
```
[the remark this answers is not in the document]
Comment (akashlevy) [reply passes/silimate/bus_rebuild.cc:219]: Fixed in df6a7c17839cd5d067cb38ad4662a79fa37358ce. The bus now reassembles `\init` from the members, setting bit `i` from the member that claimed index `i`; bits with no member of their own stay `Sx`, like any other absent bit of a sparse bus. The attribute is only written when at least one member carried one, so wires without `\init` do not gain an all-`Sx` one.

Worth noting this was a real gap rather than a copy of prior behavior: upstream `splitnets` does not carry `\init` in the forward direction either, so I had originally left it alone for symmetry. That reasoning was wrong for the join direction — splitting loses the attribute on wires that then get their own `$dff` init handling, whereas joining silently drops a reset state that nothing else re-derives.

Covered by a new case in `tests/silimate/bus_rebuild.ys` ("Per-bit init survives the merge") that asserts `init=2'b10` on the merged bus, which pins the bit order as well as the presence. Verified against the pre-fix pass that the merged bus came out with no `\init` at all.

**Comment by akashlevy** [comment]
@greptileai review
